app/uk/gov/hmrc/mobilepayments/services/OpenBankingService.scala

```diff
@@ -152,7 +152,8 @@ class OpenBankingService @Inject() (
           reference     = Some(data.originSpecificData.paymentReference.value),
           email         = email,
           emailSent     = emailSent,
-          origin        = data.originSpecificData.origin
+          origin        = data.originSpecificData.origin,
+          maybeFutureDate = Some(data.futureDatedPayment.map(_.chosenDate)).flatten
```

Comment:
Why is this wrapped in a Some? That means the potential for a Some(None) to be returned, which is never good!